userspace/libsinsp/sinsp.cpp

```diff
@@ -444,6 +450,79 @@ void sinsp::set_import_users(bool import_users)
 	m_import_users = import_users;
 }
 
+void sinsp::set_event_of_interest(const string& event_type, bool enabled)
+{
+#ifdef __linux__
+	auto machine = audit_detect_machine();
+	if (machine != -1)
+	{
+		vector<string> syscalls;
+		/*
+		 * Keep this up to date with new possible evt types that are not a syscall but
+		 * depends upon other syscalls
+		 */
+		if (event_type == "container") {
+			// TODO: better way to manage?
+			syscalls.insert(syscalls.begin(), {"chroot", "execve", "fork", "vfork", "clone"});
+		} else {
+			syscalls.push_back(event_type);
+		}
+		for (auto & syscall : syscalls)
+		{
+			/*
+			 * We need real syscall numbers because we will pass those
+			 * to driver side (kmod and eBPF)
+			 */
+			int syscall_nr = audit_name_to_syscall(syscall.c_str(), machine);
```

Comment:
Still not a fan of pulling libaudit in for this

Reply:
Would you rather go with an entire new table to map:
[ "syscall_name", SYSCALL_NR, ... (flags or whatever) ] ?
I think it would look pretty clumsy; moreover, it would add a new level of complexity when eg: adding support for new syscalls (ie: you would have to remember to add it over there).
I am not sure it would be a better solution; libaudit does all of this transparently and we are not responsible of the maintenance burden.

Reply:
Honestly, yes. Right now the process of adding a new syscall is such a complete mess (so many places need to be touched) that the way forward at this point seems to be make it even harder so that it annoys somebody enough to fix it ;)

I believe that maintaining that table (which could be just an extra field in `g_syscall_table` or `g_syscall_routing_table` (why are these two structs instead of one btw?)) is worth not having another dependency. Maybe if we used it for anything else besides this one place, it would feel less overkill.